libr/core/cfile.c

```diff
@@ -1072,15 +1072,22 @@ R_API int r_core_file_list(RCore *core, int mode) {
 		}
 		from = 0LL;
 		switch (mode) {
-		case 'j':
-			r_cons_printf ("{\"raised\":%s,\"fd\":%d,\"uri\":\"%s\",\"from\":%"
-				PFMT64d ",\"writable\":%s,\"size\":%d}%s",
-				r_str_bool (core->io->desc->fd == f->fd),
-				(int) f->fd, desc->uri, (ut64) from,
-				r_str_bool (desc->perm & R_PERM_W),
-				(int) r_io_desc_size (desc),
-				iter->n? ",": "");
+		case 'j': {  // "oj"
+			PJ * pj = pj_new ();
+			pj_o (pj);
+			pj_a (pj);
```

Comment:
an array inside an object? this doesnt seems good json to me. i think pj api should check for those invalid json serialization problems

Reply:
Oh, okay.
I just went with that, as that's what I've seen commonly implemented, and to make it similar to the current output.

What should be changed?

Reply:
{[1,2,3]} is invalid json. Just remove the pj_a